scapy/fields.py

```diff
@@ -428,16 +428,20 @@ def getfield(self, pkt, s):
 
 
 class PacketListField(PacketField):
-    __slots__ = ["count_from", "length_from"]
+    __slots__ = ["count_from", "length_from", "next_cls_cb"]
     islist = 1
-    def __init__(self, name, default, cls, count_from=None, length_from=None):
+    def __init__(self, name, default, cls=None, count_from=None, length_from=None, next_cls_cb=None):
         if default is None:
             default = []  # Create a new list for each instance
+        if not isinstance(next_cls_cb, type(None)):
```

Comment:
There is only one Pythonic way to do this, and that's: `if next_cls_cb is not None`.
Also, there is no reason to force `cls` to `Raw` here.
This whole `if ... else ...` block should be replaced by
```Python
        self.next_cls_cb = next_cls_cb
```